imap_data_access/file_validation.py

```diff
@@ -0,0 +1,226 @@
+"""Methods for managing and validating filenames and filepaths"""
+from __future__ import annotations
+
+import re
+from datetime import datetime
+from pathlib import Path
+
+import imap_data_access
+
+
+class ScienceFilepath:
```

Comment:
I was expecting Path to be capitalized because Path is from pathlib... I'm not sure there is a standard for it other than I think in other areas of the codebase we are using `file_path` with a separator indicting two words. Non-blocking suggestion though.
```suggestion
class ScienceFilePath:
```

Reply:
Sure, technically "file path" is two words so this doesn't line up with naming conventions anyway, I will change